skyportal/handlers/api/photometry.py

```diff
@@ -53,86 +60,130 @@ def post(self):
-                           zpsys=row['zpsys'],
-                           zp=row['zp'])
-
-            t = Time(row['mjd'], format='mjd')
-            converted_times.append(arrow.get(t.iso))
-
-            DBSession().add(p)
+        for i, row in df.iterrows():
+            packet = row.to_dict()
+
+            # coerce nans to nones
+            for key in packet:
+                packet[key] = nan_to_none(packet[key])
+
+            try:
+                phot = PhotometryFlux.load(packet)
+            except ValidationError as e1:
+                try:
+                    phot = PhotometryMag.load(packet)
+                except ValidationError as e2:
+                    return self.error('Invalid input format: Tried to parse '
+                                      f'{packet} as PhotometryFlux, got: '
+                                      f'"{e1.normalized_messages()}." Tried '
+                                      f'to parse {packet} as PhotometryMag, got:'
+                                      f' "{e2.normalized_messages()}."')
+
+            phot.packet = packet
+            DBSession().add(phot)
+
+            # to set up obj link
             DBSession().flush()
-            ids.append(p.id)
-        if 'thumbnails' in data:
-            p = Photometry.query.get(ids[0])
-            for thumb in data['thumbnails']:
-                create_thumbnail(thumb['data'], thumb['ttype'], obj.id, p)
-        obj.last_detected = max(
-            converted_times
-            + [
-                obj.last_detected
-                if obj.last_detected is not None
+
+            time = arrow.get(Time(phot.mjd, format='mjd').iso)
+            phot.obj.last_detected = max(
+                time,
+                phot.obj.last_detected
+                if phot.obj.last_detected is not None
                 else arrow.get("1000-01-01")
-            ]
-        )
-        DBSession().commit()
+            )
+            ids.append(phot.id)
 
+        DBSession().commit()
         return self.success(data={"ids": ids})
 
     @auth_or_token
     def get(self, photometry_id):
-        """
-        ---
-        description: Retrieve photometry
-        parameters:
-          - in: path
-            name: photometry_id
-            required: true
-            schema:
-              type: integer
-        responses:
-          200:
-            content:
-              application/json:
-                schema: SinglePhotometry
-          400:
-            content:
-              application/json:
-                schema: Error
-        """
+        # Moved to bottom of page as f-string
+
         phot = Photometry.query.get(photometry_id)
         if phot is None:
             return self.error('Invalid photometry ID')
         # Ensure user/token has access to parent source
         _ = Source.get_if_owned_by(phot.obj_id, self.current_user)
 
-        return self.success(data=phot)
+        # get the desired output format
```

Comment:
The below logic should also be applied to `SourcePhotometryHandler` (https://github.com/skyportal/skyportal/blob/94a7c7ba3638a4584afc6d394265e3bccb579ce9/skyportal/handlers/api/source.py#L384) which handles the fetching of all of a source's photometry.  It should be factored out into a function that both this and the other handler call so as to avoid code duplication.  That can be added to this PR or added in an upcoming PR, whichever you prefer.

Reply:
Also, should any of this also apply to the data used in `plot`?

Reply:
The data in `plot` is OK, it's all already standardized (stored in microJanskies).

Reply:
First comment fixed in 3f2936b